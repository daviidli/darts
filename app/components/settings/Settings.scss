.settings {
	.title {
		font-weight: 300;
		font-size: 2.5em;
		padding: 1em;
		padding-bottom: 0.5em;
	}

	.select {
		width: 25em;
	}
}
