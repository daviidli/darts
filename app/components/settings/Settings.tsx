import React, { useState, useEffect } from 'react';
// import SerialPort from 'serialport';

const Settings = () => {
	return <div>hello world</div>;
};

export default Settings;
