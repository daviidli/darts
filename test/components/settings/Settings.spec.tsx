import { spy, stub } from 'sinon';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import SerialPort from 'serialport';
import Settings, { Props } from '../../../app/components/settings/Settings';

Enzyme.configure({ adapter: new Adapter() });

const mockPush = spy();
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useHistory: () => ({
		push: mockPush
	})
}));

jest.mock('../../../app/containers/SettingsSlider', () => () => (
	<div className="SettingsSlider" />
));
jest.mock('@material-ui/core/Select', () => props => {
	const { value, onChange, label, children } = props;
	return (
		<>
			<div className="selectMock" onChange={onChange}>
				{value}
			</div>
			<div className="lableMock">{label}</div>
			<div className="childrenMock">{children}</div>
		</>
	);
});

const initialProps: Props = {
	serialPort: 'port1',
	setSerialPort: () => ({})
};

const setup = (props: Props = initialProps) => {
	const wrapper = mount(<Settings {...props} />);
	return {
		wrapper,
		select: wrapper.find('.selectMock'),
		portsList: wrapper.find('.childrenMock'),
		button: wrapper.find('.button')
	};
};

describe('Settings component', () => {
	it('should match snapshot', () => {
		const tree = renderer.create(<Settings {...initialProps} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	// it('should fetch available ports from SerialPort', () => {
	// 	jest.useFakeTimers();
	// 	stub(SerialPort, 'list').returns(
	// 		Promise.resolve([
	// 			{ path: 'port123', manufacturer: '123', serialNumber: '123' }
	// 		])
	// 	);

	// 	const { portsList } = setup();

	// 	return new Promise(resolve => {
	// 		setImmediate(() => {
	// 			expect(portsList.length).toBe(1);
	// 			expect(portsList.at(0).text()).toMatch('port123 - (123)');
	// 			resolve();
	// 		});
	// 	});
	// });
});
