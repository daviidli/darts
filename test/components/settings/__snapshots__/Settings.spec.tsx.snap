// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component should match snapshot 1`] = `
<div
  className="MuiGrid-root settings MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root gridItem MuiGrid-item"
  >
    <p
      className="MuiTypography-root title MuiTypography-body1"
    >
      Settings
    </p>
  </div>
  <div
    className="MuiGrid-root gridItem MuiGrid-item"
  >
    <div
      className="MuiFormControl-root gridItem"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
      >
        Serial Port
      </label>
      <div
        className="selectMock"
        onChange={[Function]}
      >
        port1
      </div>
      <div
        className="lableMock"
      >
        Serial Port
      </div>
      <div
        className="childrenMock"
      >
        <li
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="menuitem"
          tabIndex={-1}
          value=""
        >
          <em>
            None
          </em>
        </li>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root gridItem MuiGrid-item"
  >
    <div
      className="SettingsSlider"
    />
  </div>
  <div
    className="MuiGrid-root gridItem MuiGrid-item"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Back
      </span>
    </button>
  </div>
</div>
`;
